### PR TITLE
Add more metadata on query schema deploy create case

### PR DESCRIPTION
**Pull request by ANich**
Currently, some metadata (friendly name, description, labels) isn't attached to new or existing derived tables in BigQuery. This PR makes it so new tables get metadata attached when created via `bqetl query schema deploy`. 

I'll add an inline question about whether we should update metadata on every schema deploy rather than just on create. 

Also a few cleanup tasks like closing temp file.

---
Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title).
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated.
- [ ] When adding a new derived dataset, ensure that data is not available already (fully or partially) and recommend extending an existing dataset in favor of creating new ones. Data can be available in the [bigquery-etl repository](https://github.com/mozilla/bigquery-etl), [looker-hub](https://github.com/mozilla/looker-hub) or in [looker-spoke-default](https://github.com/mozilla/looker-spoke-default/tree/e1315853507fc1ac6e78d252d53dc8df5f5f322b).

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

### CODE REVIEW

File: bigquery_etl/cli/query.py
Comment (ANich) [inline bigquery_etl/cli/query.py:1528]: We could `_attach_metadata` on every schema deploy here actually. This way changes in `metadata.yaml` would be reflected in BigQuery on `bqetl query schema deploy`. I'm hesitating because this actually seems like something that should be done on every post-merge like the way we do view deploys.

I know we also have this: https://github.com/mozilla/bigquery-etl/blob/a190e1826498da6c898cf2910af59f7a2e8f64f2/bigquery_etl/metadata/publish_metadata.py#L32 but I'm not sure when/where that gets used?

File: bigquery_etl/cli/query.py
Comment (kik-kik) [inline bigquery_etl/cli/query.py:1523]:
```suggestion
        if not table.created:
```

**Review by curtismorales — APPROVED** [review]
I'm new to this code but this read very clean, I was able to follow the changes and this all looks right to me.

I think this is good to go, but if you want to also wait for someone more familiar with the code to review, that's fine too

File: bigquery_etl/cli/query.py
Comment (kik-kik) [inline bigquery_etl/cli/query.py:1548]: Is the `metadata.bigquery` necessary here?

File: bigquery_etl/cli/query.py
Comment (kik-kik) [inline bigquery_etl/cli/query.py:1535]: this is actually doing network calls to bq?

File: bigquery_etl/cli/query.py
Comment (kik-kik) [inline bigquery_etl/cli/query.py:1532]: Could you add a quick docstring here explaining what this does? On first look it seemed like this function does not do anything. But it looks like it actually might be interacting with BQ API.

File: bigquery_etl/cli/query.py
Comment (kik-kik) [inline bigquery_etl/cli/query.py:1510]: Looks like it might be a short circuit in the process. No need to proceed forward if an error is detected?

**Review by kik-kik — APPROVED** [review]

File: bigquery_etl/cli/query.py
Comment (ANich) [reply bigquery_etl/cli/query.py:1548]: Yup, it's optional: https://github.com/mozilla/bigquery-etl/blob/f6b20988b5f19c9ce5ce04eda405a7bae89e6ff7/bigquery_etl/metadata/parse_metadata.py#L131

File: bigquery_etl/cli/query.py
Comment (ANich) [reply bigquery_etl/cli/query.py:1538]: Yup, it's optional: https://github.com/mozilla/bigquery-etl/blob/f6b20988b5f19c9ce5ce04eda405a7bae89e6ff7/bigquery_etl/metadata/parse_metadata.py#L131

File: bigquery_etl/cli/query.py
Comment (ANich) [reply bigquery_etl/cli/query.py:1535]: Nope, no network calls until `client.update_table` or `client.create_table` is called

File: bigquery_etl/cli/query.py
Comment (ANich) [reply bigquery_etl/cli/query.py:1532]: No interactions with the API, it's attaching metadata from the file query file's metadata to the table object. I'll add a docstring

**Pull request merged**
